client/app/pages/dashboards/hooks/utils.test.js

```diff
@@ -194,3 +195,56 @@ describe("filterWidgets", () => {
     expect(dashboardAllWidgets).toStrictEqual(result);
   });
 });
+
+describe("calculateLayoutsOrder", () => {
+  test("should return layouts ordered by row and col", () => {
+    const layouts = {
+      "10": { col: 0, row: 0, sizeX: 3, sizeY: 8 },
+      "11": { col: 0, row: 8, sizeX: 3, sizeY: 8 },
+      "12": { col: 0, row: 16, sizeX: 3, sizeY: 4 },
+    };
```

Comment:
Please change the order to make sure that the tests are working. Now your code could just return the keys of the layouts and it would work.

Reply:
It was working even after i changed the tests and added a new one with more lines, but I updated the code in [this commit](https://github.com/metr-systems/redash/pull/9/commits/8f1b428598fbaed0d50a808f1648e4e8c60962d1) to be more readable because i also did not feel comfortable with the previous code. Thank you for pointing it out